A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_Init.js

```diff
@@ -0,0 +1,259 @@
+                mcdu.simbrief["estZfw"] = data.weights.est_zfw;
+                mcdu.simbrief["costIndex"] = data.general.costindex;
+                mcdu.simbrief["navlog"] = data.navlog.fix;
+                mcdu.simbrief["icao_airline"] = typeof data.general.icao_airline == 'string' ? data.general.icao_airline : "";
+                mcdu.simbrief["flight_number"] = data.general.flight_number;
+
+                if (mcdu.page.Current == mcdu.page.AOCInit) {
+                    CDUAocInit.ShowPage(mcdu);
+                }
+                CDUAocInit.showStatus(mcdu, "DONE");
+            })
+            .catch(_err => {
+                console.log(_err.message);
+                if (mcdu.page.Current == mcdu.page.AOCInit) {
+                    CDUAocInit.showStatus(mcdu, "ERROR");
+                }
+            });
+    }
+
+    /**
+     * There are two uplink requests that are made at the same time:
+     * - AOC ACT F-PLN
+     * - PERF DATA
+     */
+    static insertUplink(mcdu) {
+        const fromTo = `${mcdu.simbrief["originIcao"]}/${mcdu.simbrief["destinationIcao"]}`;
+        const cruiseAltitude = mcdu.simbrief["cruiseAltitude"];
+        const costIndex = mcdu.simbrief["costIndex"];
+        const fltNbr = `${mcdu.simbrief.icao_airline}${mcdu.simbrief.flight_number}`;
+
+        const uplinkInProg = "UPLINK INSERT IN PROG";
+        const aocActFplnUplink = "AOC ACT F-PLN UPLINK";
+        const perfDataUplink = "PERF DATA UPLINK";
+
+        mcdu.showErrorMessage(uplinkInProg);
+
+        /**
+         * AOC ACT F-PLN UPLINK
+         * TODO:
+         * - ROUTE
+         * - ALTN
+         */
+        mcdu.tryUpdateFromTo(fromTo, (result) => {
+            if (result) {
+                CDUPerformancePage.UpdateThrRedAccFromOrigin(mcdu);
+                if (mcdu.page.Current == mcdu.page.InitPageA) {
+                    CDUInitPage.ShowPage1(mcdu);
+                }
+            }
+        });
+        mcdu.updateFlightNo(fltNbr, (result) => {
+            if (result) {
+                if (mcdu.page.Current == mcdu.page.InitPageA) {
+                    CDUInitPage.ShowPage1(mcdu);
+                }
+            }
+        });
+
+        setTimeout(async () => {
+            await CDUAocInit.buildRouteFromNavlog(mcdu);
+            mcdu.showErrorMessage(aocActFplnUplink);
+        }, 10000); // Fake 10s delay for perf "calculations"
+
+        /**
+         * PERF DATA UPLINK
+         * TODO:
+         * - CRZ TEMP
+         * - BLOCK
+         * - ?
+         */
+        setTimeout(() => {
+            if (mcdu.setCruiseFlightLevelAndTemperature(cruiseAltitude)) {
+                mcdu.showErrorMessage(perfDataUplink);
+            }
+            if (mcdu.tryUpdateCostIndex(costIndex)) {
+            }
+            if (mcdu.page.Current == mcdu.page.InitPageA) {
+                CDUInitPage.ShowPage1(mcdu);
+            }
+        }, 3000); // Fake 3s delay for perf "calculations"
```

Comment:
we have central functions for that, see #1834. I recommend using one of them or adding a new one.